chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/PendingSurveillanceManagerImpl.java

```diff
@@ -0,0 +1,728 @@
+            + " as part of the replace operation failed", ex);
+        }
+
+        CertifiedProductSearchDetails afterCp =
+                cpDetailsManager.getCertifiedProductDetails(survToInsert.getCertifiedProduct().getId());
+
+        activityManager.addActivity(ActivityConcept.ACTIVITY_CONCEPT_CERTIFIED_PRODUCT, afterCp.getId(),
+                "Surveillance upload was confirmed for certified product " + afterCp.getChplProductNumber(),
+                beforeCp, afterCp);
+
+        // query the inserted surveillance
+        Surveillance result = getSurveillanceById(insertedSurv);
+        return result;
+    }
+
+    private SurveillanceUploadResult processUploadAsJob(final String data)
+            throws EntityCreationException, EntityRetrievalException {
+        SurveillanceUploadResult result = new SurveillanceUploadResult();
+
+        //figure out the user
+        UserDTO currentUser = null;
+        try {
+            currentUser = userManager.getById(Util.getCurrentUser().getId());
+        } catch (final UserRetrievalException ex) {
+            LOGGER.error("Error finding user with ID " + Util.getCurrentUser().getId() + ": " + ex.getMessage());
+            result.setJobStatus(SurveillanceUploadResult.UNAUTHORIZED);
+            return result;
+        }
+        if (currentUser == null) {
+            LOGGER.error("No user with ID " + Util.getCurrentUser().getId() + " could be found in the system.");
+            result.setJobStatus(SurveillanceUploadResult.UNAUTHORIZED);
+            return result;
+        }
+
+        JobTypeDTO jobType = null;
+        List<JobTypeDTO> jobTypes = jobManager.getAllJobTypes();
+        for (JobTypeDTO jt : jobTypes) {
+            if (jt.getName().equalsIgnoreCase(allowedJobType.getName())) {
+                jobType = jt;
+            }
+        }
+
+        JobDTO toCreate = new JobDTO();
+        toCreate.setData(data);
+        toCreate.setUser(currentUser);
+        toCreate.setJobType(jobType);
+        JobDTO insertedJob = jobManager.createJob(toCreate);
+        JobDTO createdJob = jobManager.getJobById(insertedJob.getId());
+
+        try {
+            boolean isStarted = jobManager.start(createdJob);
+            if (!isStarted) {
+                result.setJob(new Job(createdJob));
+                result.setJobStatus(SurveillanceUploadResult.NOT_STARTED);
+                return result;
+            } else {
+                createdJob = jobManager.getJobById(insertedJob.getId());
+            }
+        } catch (final EntityRetrievalException ex) {
+            LOGGER.error("Could not mark job " + createdJob.getId() + " as started.");
+            result.setJob(new Job(createdJob));
+            result.setJobStatus(SurveillanceUploadResult.ERROR);
+            return result;
+
+        }
+
+        // query the now running job
+        result.setJob(new Job(createdJob));
+        result.setJobStatus(SurveillanceUploadResult.SUCCESS);
```

Comment:
Just making sure this is being interpreted as "the job was started successfully" and not "the job was completed successfully"

Reply:
Addressed.